migration/migration/002-users.sql

```diff
@@ -64,3 +64,11 @@ CALL _prom_catalog.execute_everywhere('grant_prom_reader_prom_writer',$ee$
     END
     $$;
 $ee$);
+
+GRANT EXECUTE ON FUNCTION _prom_ext.num_cpus() TO prom_reader;
```

Comment:
can we colocate these grants with the function definitions somehow?

Reply:
Possibly, but it would require restructuring things quite a bit. Currently, the emitted sql schema file contains (in this order):

- pgx idempotent content
- hand-written migration content
- hand-written idempotent content

This means that at the point that the function is defined (pgx idempotent content) the users for the grants are not present: they are only added in this file.

We will probably have to rethink some of this structure anyway when addressing the security concerns, so perhaps we could punt on that until then? I'll add a todo to address later.

Reply:
Punting on this for now is fine. Just one other idea to consider - move this to an idempotent instead of migration script